src/Culling/ray.ts

```diff
@@ -653,15 +659,30 @@ Scene.prototype.createPickingRayInCameraSpaceToRef = function (x: number, y: num
     return this;
 };
 
+Scene.prototype._internalPickForMesh = function (pickingInfo: Nullable<PickingInfo>, rayFunction: (world: Matrix) => Ray, mesh: AbstractMesh, world: Matrix, fastCheck?: boolean, onlyBoundingInfo?: boolean, trianglePredicate?: TrianglePickingPredicate) {
+    let ray = rayFunction(world);

```

Comment:
I think this function should take a thinInstanceIndex (-1 by default) that it copies into the returned `PickingInfo` structure, so that the user can know which thin instance is intersected.

Reply:
yeah I debated with myself a lot about that...but you right I'll add it